### DISCUSSION TITLE
Automatic import of #ID_NUMBER into node main field

### DISCUSSION THREAD

**Discussion by Engesser** [freeplane/freeplane#676] [Help]
hello,

as I am always integrating the #ID_NUMBER into the node main field, can I adjust the program to automatically add the #ID_NUMBER into node main field or DETAILS or NOTICE? Or do I have to use a script?

Thanks for answering,

Karl

**Comment by macmarrum** [comment]
I know only of the option `Preferences…->Appearance->Status line->Display node ID`. Anything else requires a script. 
I often use a ChangeListener to do something automatically - in your case it would add #ID_NUMBER to node core or details or note. Let me know if you need help setting it up.

**Reply by Engesser** [reply]
hello macmarrum, thank you for your helpful comment. I am not really an expert in scripting nor do I know what a "ChangeListener" is, so I will not make any progress without help? I hoped that Dimitry could add this feature into the menu at some place? I will ask for that in a separate discussion?
Up to now I am using an F6 button to copy the ID and then manually add it to the node.
Greetings

Karl

**Reply by macmarrum** [reply]
Hello Karl

Sure thing. I have a ChangeLinstener script handy and will make it available to you.
Just let me know if you'd like #ID_NUMBER added to every new node in Freeplane (any mindmap) or only for selected mindmaps.

Regards
Mac

**Reply by Engesser** [reply]
Hello Mac,

I am not in a hurry. But may be I missed an answer from you concerning the automatic ID-Import (ChangeLinstener script)?

many greetings

Karl

**Reply by macmarrum** [reply]
Hi Karl

You can use my listener to have ID_NUMBER in every new node of your mindmap.

1. Download [MacmarrumChangeListenerUtils.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/lib/MacmarrumChangeListenerUtils.groovy) and place it in `<user-dir>/lib`.
_Note: You can use `Tools->Open user directory`_
2. Install  [ScriptExecutionOnMindmapOpen-2022.09.26.addon.mm](https://github.com/macmarrum/freeplane-ScriptExecutionOnMindmapOpen/releases/download/v2022.09.26/ScriptExecutionOnMindmapOpen-2022.09.26.addon.mm)
3. In the root node of your mindmap
   1. Add an attribute named `scriptOnMapOpen` with the value `MacmarrumChangeListenerUtils.toggleMapChangeListener(node)`
   2. Add an attribute named `MacmarrumFeatures` with the value `ID_ON_CREATE`
   3. Add an attribute named `IdLocation` with the value `CORE` or `DETAILS` or `NOTE` depending where you want ID_NUMBER to be located
4. Restart your Freeplane

![2022-09-30_14,07,27](https://user-images.githubusercontent.com/7197131/193266309-82b8509d-b7a1-47ca-bf0f-3303d798a59a.png)


Regards
Mac
